build_tools/github_actions/python_to_cp_version.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+"""Determines part of the path to the Python interpreter in a manylinux image.
+
+Example usage:
+
+    python python_to_cp_version.py --python-version 3.12
+
+  The following string is appended to the file specified in the "GITHUB_ENV"
+  environment variable:
+
+    cp_version=cp312-cp312
+
+Writing the output to the "GITHUB_ENV" file can be suppressed by passing
+`--no-write-env-file`.
+"""
+
+import argparse
+import os
+import re
+import sys
+
+
+def is_version(version) -> bool:
+    return re.match(r"^([1-9])\.((0|([1-9][0-9]*)))(t)?$", version) is not None
```

Comment:
I wonder if there's a built in way to parse, like using https://packaging.pypa.io/en/stable/version.html ... probably better to just be explicit here given we care about the _python_ version, not a package version though.

Reply:
I checked https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions, which is what `packaging` would use and that's much broader that what we want to allow.